docs/features.md

```diff
@@ -18,6 +18,8 @@ You can watch demos for some of these features [below](#demos).
 - [Call hierarchy support](#call-hierarchy)
 - [Qualify names from an import declaration](#qualify-imported-names) in your code
 - [Suggest alternate numeric formats](#alternate-number-formatting)
+- [Renaming](#renaming) definitions automatically across modules of *the same component* (library, executable, test suite, benchmark).
```

Comment:
This is not correct. Renaming works across all modules that have been indexed (in this or previous sessions).

Reply:
Aha, thanks, please, could you note in which cases the lack of multicomponent support could affect the plugin behaviour? does it work for the components currently loaded? so if you open files from components forcing their loading, the renaming would work across those loaded components (and no others)?

Reply:
> could you note in which cases the lack of multicomponent support could affect the plugin behaviour? 

I already said: the plugin will rename all occurrences that have been indexed in this or previous IDE sessions.

One example where the plugin would fail to find some occurrences is the first time a project has been loaded in the IDE *ever*, and only a subset of the components are loaded.

> does it work for the components currently loaded? 

It works for all the components indexed, whether they are loaded or not.

> so if you open files from components forcing their loading, the renaming would work across those loaded components (and no others)?

If you open a file from a component, that will cause the component to be fully indexed. So assuming that you wait until indexing is done, the plugin will work for that component.

Reply:
Ok so indexed as indexed in hiedb, i guess, thanks for the clarification
Afaiu, as we have the option `haskell.checkProject`, hls by default would index all the project opening any file from it, right?
If it is set to false the best way to ensure the last version of files are indexes would open all files where the symbol to berenamed i present.

Reply:
I think that the caveat should be that it only works for single component projects. It does not work for a single component within a multi-component project.

Indexing all components of files where the symbol is referenced before renaming should work (I did this manually when recording the [example gif](https://github.com/haskell/haskell-language-server/pull/2108#issuecomment-918075894)).
We could grep the workspace for the symbol to do this (since reference finding also won't work yet), however, we would need to be careful not to index files that don't have a proper cradle.

Reply:
> Afaiu, as we have the option haskell.checkProject, hls by default would index the entire the project opening any file from it, right?

If by project you mean component, then yes.

> If it is set to false the best way to ensure the last version of files are indexes would open all files where the symbol to berenamed i present.

If it is set to false, components won't be indexed in full on open. So you could just open all the files in the workspace.